tutorial/02_segmentation_and_measuring.md

```diff
@@ -1,494 +1,598 @@
-        image[circle] = 255,255,0
- 
-    return image
-```
+- Read the Cellpose documentation example notebook [here](https://cellpose.readthedocs.io/en/latest/notebook.html).
+- Index the middle plane of your cells3d image.
+- Use [`np.transpose`](https://numpy.org/doc/stable/reference/generated/numpy.transpose.html) to move the channel axis to the final position in the array.
+- Call cellpose with the appropriate channel settings (carefully read the example in the Cellpose documentation).
+- Display the two image channels and the segmentation masks overlaid in napari.
 
 ```{code-cell} ipython3
-:tags: [raises-exception, remove-output]
-
-# All edges are drawn with threshold at infinity
-edges_drawn_all = display_edges(astronaut_felzenszwalb_colored, rag, np.inf)
-image_show(edges_drawn_all);
+# solution here
 ```
 
-Try a range of thresholds out, see what happens.
+## A napari plugin
 
-```{code-cell} ipython3
-:tags: [raises-exception, remove-output]
++++
 
-threshold = ... # Experiment
+Napari UI plugins are namespaces (e.g. modules) containing functions that implement one of several napari *hook specifications* — that is, function signatures matching a certain pattern. For more details, see the napari plugins documentation:
 
-edges_drawn_few = display_edges(astronaut_felzenszwalb_colored, rag, threshold)
-image_show(edges_drawn_few);
-```
+https://napari.org/plugins/stable/index.html
 
-#### Finally, cut the graph
+Today, we will implement a plugin to separate labels based on points, as we did above.
 
-Once you are happy with the (dis)connected regions above, the graph can be cut to merge the regions which are still connected.
+Here's a simple little plugin:
 
 ```{code-cell} ipython3
-:tags: [raises-exception, remove-output]
+import napari
+from magicgui import magic_factory
+from napari_plugin_engine import napari_hook_implementation
 
-final_labels = graph.cut_threshold(astronaut_felzenszwalb + 1, rag, threshold)
-final_label_rgb = color.label2rgb(final_labels, astronaut, kind='avg')
 
-image_show(final_label_rgb);
-```
+@magic_factory(result_widget=True)
+def f(x: int, y: int) -> int:
+    return x + y
 
-How many regions exist now?
 
-```{code-cell} ipython3
-:tags: [raises-exception, remove-output]
+@magic_factory(result_widget=True)
+def f2(x: int, y: int) -> int:
+    return x - y
 
-np.unique(final_labels).size
+@napari_hook_implementation
+def napari_experimental_provide_dock_widget():
+    return [f, f2]
 ```
 
-## Exercise: Cat picture
-
-The data directory also has an excellent image of Stéfan's cat, Chelsea.  With what you've learned, can you segment the cat's nose?  How about the eyes?  Why is the eye particularly challenging?
-
-Hint: the cat's nose is located close to [240, 270] and the right eye center is near [110, 172] in row, column notation.
+Now we register the plugin. **Note: because of a bug, you will need to Kernel -> Restart Kernel if you have created a viewer and closed it in this session.**
```

Comment:
Is this bug still present? I think it works on my machine? The only warnings that are shown seem to pertain to little-calculator not yet being registered / or registered twice.

Reply:
I'm not 100%... Maybe we can change "will" to "may".

```suggestion
Now we register the plugin. **Note: because of a bug, you may need to Kernel -> Restart Kernel if you have created a viewer and closed it in this session.**
```